File: boj7/2745.cc
// 진법 변환
// https://www.acmicpc.net/problem/2745

#include <iostream>

using namespace std;

string N;
int B;

int main() {
  cin >> N >> B;

  int ans = 0;
  for (int i = 0; i < N.size(); i++) {
    char digit_ch = N[i];
    int digit;
    if (digit_ch >= '0' && digit_ch <= '9') {
      digit = digit_ch - '0';
    } else {
      digit = 10 + (digit_ch - 'A');
    }
    ans = ans * B + digit;
  }
  printf("%d\n", ans);
  return 0;
}

File: past_archive/23boj_py/2745.py
# 진법 변환
# https://www.acmicpc.net/problem/2745

N, B = input().split()
B = int(B)

num = 0
exp = 0
for digit in N[::-1]:
    if digit >= '0' and digit <= '9':
        digit = ord(digit) - ord('0')
    else:
        digit = ord(digit) - ord('A') + 10
    num += pow(B, exp) * digit
    exp += 1

print(num)
